packages/vm/src/runBlock.ts

```diff
@@ -114,8 +114,8 @@ export default async function runBlock(this: VM, opts: RunBlockOpts): Promise<Ru
    */
   await this._emit('beforeBlock', block)
 
-  if (this._hardforkByBlockNumber) {
-    this._common.setHardforkByBlockNumber(block.header.number.toNumber())
+  if (this._hardforkByBlockNumber || this._hardforkByTD) {
+    this._common.setHardforkByBlockNumber(block.header.number.toNumber(), this._hardforkByTD)
```

Comment:
we don't need `number.toNumber()` here anymore since we expanded the input params for `setHardforkByBlockNumber` it accepts BN now.

```suggestion
    this._common.setHardforkByBlockNumber(block.header.number, this._hardforkByTD)
```